userguide/sharing.rst

```diff
@@ -527,10 +527,85 @@ the configuration screen is open for the *nfs_share1* share.
    NFS Share Settings
 
 
+Remember these points when creating NFS shares:
+
+#.  Clients will specify the :guilabel:`Path` when mounting the share.
+
+#.  The :guilabel:`Maproot` and :guilabel:`Mapall` options are
+    exclusive, meaning only one can be used as the |web-ui| does
+    not allow both. The :guilabel:`Mapall` options supersede the
+    :guilabel:`Maproot` options. To restrict only the *root* user
+    permissions, set the :guilabel:`Maproot` option. To restrict
+    permissions of all users, set the :guilabel:`Mapall` options.
+
+#.  Each volume or dataset is considered to be its own filesystem and
+    individual NFS shares cannot cross filesystem boundaries. Sharing
+    multiple directories with an additional paths is only allowed within
+    a single volume or dataset.
+
+#.  The network and host must be unique per share and filesystem or
+    directory. Because :file:`/etc/exports` does not act like an ACL,
+    the rule to apply is undefined among overlapping networks or when
+    using the same share with multiple hosts.
+
+#.  The :guilabel:`All dirs` option can only be used once per share per
+    filesystem.
+
+
+To better understand these restrictions, consider a scenario where
+there are:
+
+* two networks, *10.0.0.0/8* and *20.0.0.0/8*
+
+* a ZFS volume named :file:`volume1` with 2 datasets named
+  :file:`dataset1` and :file:`dataset2`
+
+* :file:`dataset1` contains directories named :file:`directory1`,
+  :file:`directory2`, and :file:`directory3`
+
+Because of restriction #3, an error is shown when trying to create one
+NFS share like this:
+
+* :guilabel:`Authorized networks` set to *10.0.0.0/8 20.0.0.0/8*
+
+* :guilabel:`Path` set to :file:`/mnt/volume1/dataset1` and
+  :file:`/mnt/volume1/dataset1/directory1`
+
+The correct method to configure this share is to set the
+:guilabel:`Path` to :file:`/mnt/volume1/dataset1` and set
+:guilabel:`All Directories`. This allows the client to also mount
+:file:`/mnt/volume1/dataset1/directory1` when
+:file:`/mnt/volume1/dataset1` is mounted.
+
+Additional paths are used to configure the share to point to separate
+directories within a single pool or dataset filesystem. Two paths within
+a single share are needed to share
+:file:`/mnt/volume1/dataset1/directory1` and
+:file:`/mnt/volume1/dataset1/directory2`, but not
+:file:`/mnt/volume1/dataset1/directory3`.
```

Comment:
Not clear what this example is saying.  It looks like it's just saying that you need two paths to share two directories, but not sure what the third one is saying.  Why a separate third one?  Could that done by showing a single path is needed to share directory1, but two are needed to share directory1 and directory2?